GoogleDataTransport/GDTCORLibrary/GDTCORPlatform.m

```diff
@@ -118,6 +121,51 @@ GDTCORNetworkMobileSubtype GDTCORNetworkMobileSubTypeMessage() {
 #endif
 }
 
+void GDTCOREncodeArchive(id<NSSecureCoding> obj, NSString *archivePath, NSError *_Nullable *error) {
+#if (defined(__IPHONE_11_0) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 110000) || \
+    (defined(__MAC_10_13) && MAC_OS_X_VERSION_MAX_ALLOWED >= 101300) ||      \
+    (defined(__TVOS_11_0) && __TV_OS_VERSION_MAX_ALLOWED >= 110000)
+  if (@available(macOS 10.13, iOS 11.0, tvOS 11.0, *)) {
+    NSData *data = [NSKeyedArchiver archivedDataWithRootObject:obj
+                                         requiringSecureCoding:YES
+                                                         error:error];
+    BOOL result = [data writeToFile:archivePath atomically:YES];
+    if (!result) {
+    }
+    GDTCORLogDebug(@"Attempt to write archive. successful:%@ path:%@", result, archivePath);
+  } else {
+#elif !TARGET_OS_MACCATALYST && !TARGET_OS_WATCH
+  BOOL result [NSKeyedArchiver archiveRootObject:obj toFile:archivePath];
```

Comment:
BTW, the old methods throw an exception instead of returning an error if something goes wrong. It may be useful to catch the exception and convert it into an error.